False positive property-with-parameters for an @abstractproperty setter

### Steps to reproduce
1. Define the following code with an abstract setter:
```python
"""MyModule."""
from abc import ABC, abstractproperty


class MyClassBase(ABC):
    """MyClassBase."""

    @abstractproperty
    def example(self):
        """Getter."""

    @abstractproperty
    @example.setter
    def example(self, value):
        """Setter."""
```
2. Run pylint on the file

### Current behavior
The following problem is reported for the abstract setter:
```
example.py:14:4: R0206: Cannot have defined parameters for properties (property-with-parameters)
```
Removing the `@abstractproperty` on line 12 makes the `property-with-parameters` problem disappear.

### Expected behavior
It should not report any problem for the definition of the abstract setter as it's correct that it has a parameter to set the value.

### pylint --version output
```console
$ pylint --version
pylint 2.5.2
astroid 2.4.1
Python 3.7.7 (default, Mar 10 2020, 15:43:03)
[Clang 11.0.0 (clang-1100.0.33.17)]
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,10 @@ What's New in Pylint 2.5.3?
 
 Release date: TBA
 
+* `property-with-parameters` properly handles abstract properties
+
+  Close #3600
+
 
 What's New in Pylint 2.5.2?
 ===========================

--- a/pylint/checkers/classes.py
+++ b/pylint/checkers/classes.py
@@ -1109,7 +1109,12 @@ a metaclass class method.",
             )
 
     def _check_property_with_parameters(self, node):
-        if node.args.args and len(node.args.args) > 1 and decorated_with_property(node):
+        if (
+            node.args.args
+            and len(node.args.args) > 1
+            and decorated_with_property(node)
+            and not is_property_setter(node)
+        ):
             self.add_message("property-with-parameters", node=node)
 
     def _check_invalid_overridden_method(self, function_node, parent_function_node):

--- a/tests/functional/p/property_with_parameters.py
+++ b/tests/functional/p/property_with_parameters.py
@@ -1,7 +1,21 @@
 # pylint: disable=missing-docstring, too-few-public-methods
+from abc import ABCMeta, abstractproperty
 
 
 class Cls:
     @property
     def attribute(self, param, param1): # [property-with-parameters]
         return param + param1
+
+
+class MyClassBase(metaclass=ABCMeta):
+    """MyClassBase."""
+
+    @abstractproperty
+    def example(self):
+        """Getter."""
+
+    @abstractproperty
+    @example.setter
+    def example(self, value):
+        """Setter."""

--- a/tests/functional/p/property_with_parameters.txt
+++ b/tests/functional/p/property_with_parameters.txt
@@ -1,1 +1,1 @@
-property-with-parameters:6:Cls.attribute:Cannot have defined parameters for properties
+property-with-parameters:7:Cls.attribute:Cannot have defined parameters for properties

